server/webserver/queries.py

```diff
@@ -0,0 +1,16 @@
+from webserver.extensions import db
+from webserver.models import User
+
+
+def create_user(email, name, hash, authentication_method):
+    user = User(
+        email=email, name=name, hash=hash, authentication_method=authentication_method
+    )
+    db.session.add(user)
+    db.session.commit()
+    return
+
+
+def get_user(name):
+    user = db.session.execute(db.select(User).where(name=name))
```

Comment:
will names be unique? or should we use email or ids?

Reply:
also has not been discussed, not sure if we want to display a user's full name or first and last name or just a username. I think we can alternatively require email for login. Any uniqueness can be enforced in the model definitions.

Reply:
Do we need to keep first and last names? i think just email and password no? Theres no social aspect for this right now, and i believe no user profiles in scope

Reply:
This is from the figma, I will message the chat to see what the plan is and if you're right we can just modify the schema
![image](https://user-images.githubusercontent.com/58638902/233427479-18ce5fe6-5df0-40f4-9ef5-469986239dbf.png)

Reply:
Another note, names are not unique, so we want to use a unique identifier

Reply:
Sounds good, I will look at either email or id as a unique identifier for lookup. Any thoughts on having two separate queries for `get_user_by_email` and `get_user_by_id`? My thinking is that at the login/registration phase there's no way for the user to supply their id but when the client side is requesting information about a user I think it makes sense for them to request using the id.

Reply:
Depends on use case, i think at registratioan, we've got to validate for duplicate entities anyways, (which i think you're doing). Fetching user deatils however, w/e unique identifier we decide on whether it be id or email, should be okay--as long as that information is provided as context to the client prior to any future requests.

ie:
1. User signs in
2. server responds with authenticated user payload (email/username)
3. client stores logged in user to session/state

In this case, any requests later, if necessary (if not token), client can make requests and include user email as a param if necessary

Reply:
Currently using JWT so the login flow would be:

1. User signs in (client submits user form data)
2. The user info is verified on the server
3. Server sends token to client to use in subsequent requests

In step 3 I think we should send back additional information other than just the access_token. I think we can use this step to send the client the information it needs (id or email) in order to request specific user data.

Additionally, when Google auth is configured, the server can check the user information sent from Google and if the user already exists in the database we can send the same data back as the above method.